contracts/cryptoblades.sol

```diff
@@ -451,6 +460,60 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
         return (((attacker + 1) % 4) == defender); // Thanks to Tourist
     }
 
+    function _updatePaymentBlockHash() internal {
+        if ((mintPayments[msg.sender].count > 0) &&
+            (mintPayments[msg.sender].blockHash == 0) &&
+            (mintPayments[msg.sender].blockNumber < block.number)) {
+
+            // Payment must be recent enough that the hash is available for the payment block.
+            // Use 200 as a 'friendly' window of "You have 10 minutes."
+            if (mintPayments[msg.sender].blockNumber + 200 >= block.number) {
+                mintPayments[msg.sender].blockHash = blockhash(mintPayments[msg.sender].blockNumber);
+            }
+        }
+    }
+
+    function _discardPaymentIfExpired() internal {
+        _updatePaymentBlockHash();
+        if (mintPayments[msg.sender].count > 0) {
+            if (mintPayments[msg.sender].blockHash == 0) {
+                delete mintPayments[msg.sender];
+            }
+        }
+    }
+
+    function hasPaidForMint() public view returns(bool){
+        return (mintPayments[msg.sender].count > 0);
+    }
+
+    function payForMint(address nftAddress, uint count) public {
+        _discardPaymentIfExpired();
+
+        require(mintPayments[msg.sender].count == 0);
+
+        require(nftAddress == address(weapons));
+        _payContract(msg.sender, mintWeaponFee);
+
+        mintPayments[msg.sender].count = count;
+        mintPayments[msg.sender].nftAddress = nftAddress;
+        mintPayments[msg.sender].blockNumber = block.number;
+        mintPayments[msg.sender].blockHash = 0;
+    }
+
+    function _usePayment(address nftAddress, uint count) internal {
+        _discardPaymentIfExpired();
+
+        require(mintPayments[msg.sender].count >= count);
```

Comment:
we may want the .count check to be == instead of >=.
Imagine paying for 10 mints, then for some reason deciding to use the 2nd step for the 1x mint.

Reply:
> we may want the .count check to be == instead of >=.
> Imagine paying for 10 mints, then for some reason deciding to use the 2nd step for the 1x mint.

With >= the code already supports this scenario.  Changing it to == breaks that.